plugins/net_plugin/net_plugin.cpp

```diff
@@ -2464,16 +2512,17 @@ namespace eosio {
    }
 
    void net_plugin_impl::start_listen_loop() {
-      connection_ptr new_connection = std::make_shared<connection>();
+      connection_ptr new_connection = std::make_shared<connection>(ssl_enabled);
       new_connection->connecting = true;
       new_connection->strand.post( [this, new_connection = std::move( new_connection )](){
-         acceptor->async_accept( *new_connection->socket,
-            boost::asio::bind_executor( new_connection->strand, [new_connection, socket=new_connection->socket, this]( boost::system::error_code ec ) {
+         acceptor->async_accept( new_connection->get_socket(),
+            boost::asio::bind_executor( new_connection->strand, [new_connection, this]( boost::system::error_code ec ) {
```

Comment:
This needs to capture the shared_ptr of the socket so it maintains the lifetime of the socket.